pallets/network-score/src/tests.rs

```diff
@@ -384,3 +384,91 @@ fn revoke_rating_with_existing_rating_identifier_should_fail() {
 		);
 	});
 }
+#[test]
+fn test_revise_rating_id_already_exists() {
+	let creator = DID_00.clone();
+	let author = ACCOUNT_00.clone();
+	let message_id = BoundedVec::try_from([82u8; 10].to_vec()).unwrap();
+	let entity_id = BoundedVec::try_from([73u8; 10].to_vec()).unwrap();
+	let provider_id = BoundedVec::try_from([74u8; 10].to_vec()).unwrap();
+	let entry = RatingInputEntryOf::<Test> {
+		entity_id,
+		provider_id,
+		total_encoded_rating: 250u64,
+		count_of_txn: 7u64,
+		rating_type: RatingTypeOf::Overall,
+		provider_did: creator.clone(),
+	};
+	let entry_digest =
+		<Test as frame_system::Config>::Hashing::hash(&[&entry.encode()[..]].concat()[..]);
+	let raw_space = [2u8; 256].to_vec();
+	let space_digest = <Test as frame_system::Config>::Hashing::hash(&raw_space.encode()[..]);
+	let space_id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_digest.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+	let space_id: SpaceIdOf = generate_space_id::<Test>(&space_id_digest);
+	let auth_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_id.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+	let authorization_id: AuthorizationIdOf =
+		Ss58Identifier::create_identifier(&auth_digest.encode()[..], IdentifierType::Authorization)
+			.unwrap();
+	let id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[
+			&entry_digest.encode()[..],
+			&entry.entity_id.encode()[..],
+			&message_id.encode()[..],
+			&space_id.encode()[..],
+			&creator.clone().encode()[..],
+		]
+		.concat()[..],
+	);
+
+	let identifier =
+		Ss58Identifier::create_identifier(&(id_digest).encode()[..], IdentifierType::Rating)
+			.unwrap();
+
+	let rating_details = <RatingEntries<Test>>::get(&identifier).unwrap();
+
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+
+		assert_ok!(Space::create(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			space_digest
+		));
+		assert_ok!(Space::approve(RawOrigin::Root.into(), space_id.clone(), 3u64));
+
+		assert_ok!(Score::register_rating(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			entry.clone(),
+			entry_digest,
+			message_id.clone(),
+			authorization_id.clone(),
+		));
+
+		<MessageIdentifiers<Test>>::remove(message_id.clone(), creator.clone());
+
+		<RatingEntries<Test>>::insert(
+			&identifier,
+			RatingEntryOf::<Test> {
+				creator_id: author.clone(),
+				entry_type: EntryTypeOf::Debit,
+				reference_id: Some(identifier.clone()),
+				..rating_details
+			},
+		);
+
```

Comment:
revoke_rating will take the identifier returned (created during) register_rating, with optional new messgae_id.
When `revoke_rating` happens it creates a `debit` entry in the chain which will be useful for give a new rating through `revise_rating` to create a new `credit` entry.

Reply:
Ohk ! @vatsa287  , So i have to get the rating which was created during `register_rating` , but in my current approach , i have generated an identifier which is created from the arguments of `register_rating` function , why is that giving `RatingIdentifierAlreadyAdded` error ?

Reply:
Please paste the screenshot with the code. I'll look up and review.

Reply:
```
let id_digest = <Test as frame_system::Config>::Hashing::hash(
&[
	&entry_digest.encode()[..],
	&entry.entity_id.encode()[..],
	&message_id.encode()[..],
	&space_id.encode()[..],
	&creator.clone().encode()[..],
]
.concat()[..],);
let identifier =
       Ss58Identifier::create_identifier(&(id_digest).encode()[..], IdentifierType::Rating)
	.unwrap();
```
![Screenshot 2024-04-15 162927](https://github.com/dhiway/cord/assets/58181784/02778e08-bec8-4a22-aa9b-277f3c9d4d9c)